### PR TITLE
Only transpile files which have changed

### PR DESCRIPTION
**Pull request by xCuri0**
Reduces build time alot on lower end cpus. Checking is done by setting and comparing file date so you can still overwrite edited files if you need to

### CODE REVIEW

File: build/transpile.js [hunk 1]
```diff
@@ -924,25 +924,26 @@ class Transpiler {
...
-            const phpmtime = fs.statSync (phpFolder + filename.replace ('.js', '.php')).mtime
+            const pythonFilename = filename.replace ('.js', '.py')
+            const phpFilename = filename.replace ('.js', '.php')
+            const jsmtime = fs.statSync (jsFolder + filename).mtime.getTime ()
+            const python2mtime = fs.statSync (python2Folder + pythonFilename).mtime.getTime ()
+            const python3mtime = fs.statSync (python3Folder + pythonFilename).mtime.getTime ()
+            const phpmtime = fs.statSync (phpFolder + phpFilename).mtime.getTime ()
             const contents = fs.readFileSync (jsFolder + filename, 'utf8')
 
-            if (jsmtime.getTime() != python2mtime.getTime() || jsmtime.getTime() != python3mtime.getTime() || jsmtime.getTime() != phpmtime.getTime()) {
```
Comment (xCuri0) [inline build/transpile.js]: did you test this ? because from my testing it always thinks the file has changed if you do this

**Comment by kroitor** [comment]
@xCuri0 i'm adding a few more edits and merging it shortly, no worries, and thanks for your help!

**Comment by kroitor** [comment]
Actually, it looks like I will have to revert this change. Because after a `git clone` all mtimes on Travis are equal, and that breaks the publishing.

**Comment by kroitor** [comment]
As an alternative to it, you can `npm run transpile binance` or any other exchange, one by one.

**Comment by kroitor** [comment]
I will probably add a `--force` flag to the transpiler, so that it will always retranspile the files on travis, but will do the mtime check locally.

**Pull request merged**